dish_grpc_influx2.py

```diff
@@ -0,0 +1,330 @@
+
+    opts = dish_common.run_arg_parser(parser, need_id=True)
+
+    opts.icargs = {}
+    for key in ["url", "token", "bucket", "org", "verify_ssl"]:
+        val = getattr(opts, key)
+        if val is not None:
+            opts.icargs[key] = val
+
+    if opts.verify_ssl is not None:
+        opts.icargs["ssl"] = True
+
+    return opts
+
+
+def flush_points(opts, gstate):
+    try:
+        write_api = gstate.influx_client.write_api(write_options=WriteOptions(batch_size=len(gstate.points),
+                                                                              flush_interval=10_000,
+                                                                              jitter_interval=2_000,
+                                                                              retry_interval=5_000,
+                                                                              max_retries=5,
+                                                                              max_retry_delay=30_000,
+                                                                              exponential_base=2))
+        while len(gstate.points) > MAX_BATCH:
+            write_api.write(record=gstate.points[:MAX_BATCH], write_precision=WritePrecision.S, bucket=opts.bucket)
+            if opts.verbose:
+                print("Data points written: " + str(MAX_BATCH))
+            del gstate.points[:MAX_BATCH]
+
+        if gstate.points:
+            write_api.write(record=gstate.points, write_precision=WritePrecision.S, bucket=opts.bucket)
+            if opts.verbose:
+                print("Data points written: " + str(len(gstate.points)))
+            gstate.points.clear()
+        write_api.flush()
+        write_api.close()
+    except Exception as e:
+        dish_common.conn_error(opts, "Failed writing to InfluxDB database: %s", str(e))
+        # If failures persist, don't just use infinite memory. Max queue
+        # is currently 10 days of bulk data, so something is very wrong
+        # if it's ever exceeded.
+        if len(gstate.points) > MAX_QUEUE_LENGTH:
+            logging.error("Max write queue exceeded, discarding data.")
+            del gstate.points[:-MAX_QUEUE_LENGTH]
+        return 1
+
+    return 0
+
+
+def query_counter(opts, gstate, start, end):
+    try:
+        query_api = gstate.influx_client.query_api()
+        print(str(opts.bucket) + '  ' + str(start) + "  " + str(end))
+        result = query_api.query('''
+        from(bucket: "{0}")
+            |> range(start: {1}, stop: {2})
+            |> filter(fn: (r) => r["_measurement"] == "{3}")
+            |> filter(fn: (r) => r["_field"] == "counter")
+            |> last()
+            |> yield(name: "last")
+            '''.format(opts.bucket, str(start), str(end), BULK_MEASUREMENT)
+        )
+        if result:
+            counter = result[0].records[0]['_value']
+            timestamp = result[0].records[0]['_time']
+            timestamp = time.mktime(timestamp.timetuple())
```

Comment:
In my testing, this appeared to have converted wrong. This should be seconds since epoch as a UTC time (actually, it should be whatever time.time() returns, but I have yet to encounter a system where that wasn't UTC), and this seems to be offsetting by local time offset, instead.

While this isn't critical functionality, as it's just making sure there isn't a +/-1 second offset from prior logged bulk history values, if it's going to be included, it should be corrected. timestamp.timestamp() seems to give the correct value.